collectd/files/collectd.conf

```diff
@@ -1,16 +1,28 @@
-Hostname "{{ hostname }}"
-FQDNLookup true
+{%- from "collectd/map.jinja" import collectd_settings with context -%}
+
+#
+# DO NOT EDIT
+#
+# This file is managed by salt via {{ source }}
+# Modify the config that generates this file instead
+#
+
+Hostname "{{ salt['grains.get']('fqdn') }}"
+FQDNLookup {{ collectd_settings.FQDNLookup }}
 #BaseDir "/var/lib/collectd"
 #PluginDir "/usr/lib/collectd"
-TypesDB {% for type in types %} "{{ type }}" {% endfor %}
+
+{% for type in collectd_settings.TypesDB %}
+TypesDB "{{ type }}"
+{%- endfor %}
 #Interval 10
```

Comment:
Both are fine, I prefer shorter lines and meaningful diffs: https://collectd.org/documentation/manpages/types.db.5.shtml